MC-38903: Add more graphic example how new image resizing works (#8243)
magento/devdocs: src/guides/v2.4/config-guide/remote-storage/config-remote-storage-caching.md

---
group: configuration-guide
title: Configure caching for Remote Storage
functional_areas:
  - Configuration
  - System
  - Setup
---

> Remote storage provide a cache layer for external API calls. The caching is based on [Flysytem cache] provided adapters.

Reviewer: ```suggestion
Remote storage provides a cache layer for external API calls. The caching is based on [Flysystem cache] provided adapters.
```

## Memory cache

> Memory cache is enabled by default and provides per-request cache.

Reviewer: ```suggestion
Memory cache is enabled by default and provides per request cache.
```

## Redis cache

> A persistent cache stored in Redis and requires an active Redis service and additional configuration

Reviewer: ```suggestion
A persistent cache stored in Redis that requires an active Redis service and additional configuration
```

### Redis cache with predis extension.

{:.procedure}

> To use Redis cache with [predis/predis] extension, perform next steps:

Reviewer: ```suggestion
To use Redis cache with [predis/predis] extension, perform the next steps:
```

1. Add [predis/predis] extension to your `composer.json`

```bash
composer require predis/predis
```

> 2. Add a configuration to `app/etc/env.php` file:

Reviewer: ```suggestion
2. Add a configuration to the `app/etc/env.php` file:
```

```php
[
    'remote_storage' => [
        # Remote storage configuration section
        'cache' => [
            'adapter' => 'predis',
                'config' => [
                    'scheme' => 'tcp', # Required scheme
                    'host' => 'redis.internal', # Required hostname of Redis service
                    'port' => 6379 # Required port of Redis service
                ]
        ]
    ]
];
```

> [Flysytem cache]: https://flysystem.thephpleague.com/v1/docs/advanced/caching

Reviewer: ```suggestion
[Flysystem cache]: https://flysystem.thephpleague.com/v1/docs/advanced/caching
```
